### ISSUE TITLE
Restrictions only restrict when variable inside braces

### ISSUE THREAD

**Issue by elliots** [a8m/envsubst#9]
If this is intended, apologies... I didn't see it documented.

```golang
package main

import (
	"log"
	"github.com/a8m/envsubst"
)

func main() {
	process("Hello $name, how are you?")
	process("Hello ${name}, how are you?")
}

func process(tpl string) {
	out, err := envsubst.StringRestricted(tpl, true, true)
	if err != nil {
		panic(err)
	}
	log.Println("Shouldn't be here - " + out)
}
```

**Comment by a8m** [comment]
Thanks for reporting this issue @elliots, it looks like a bug.

I'll try to come with a fix in the new few days. but of course any contribution is welcome.

**Comment by a8m** [comment]
solved with #11
